src/test/java/io/xstefank/wildfly/bot/PRCommitCheckTest.java

```diff
@@ -0,0 +1,119 @@
+package io.xstefank.wildfly.bot;
+
+import io.quarkiverse.githubapp.testing.GitHubAppTest;
+import io.quarkiverse.githubapp.testing.GitHubAppTesting;
+import io.quarkus.test.junit.QuarkusTest;
+import io.xstefank.wildfly.bot.utils.GitHubJson;
+import io.xstefank.wildfly.bot.utils.Util;
+import org.junit.jupiter.api.Test;
+import org.kohsuke.github.GHEvent;
+import org.kohsuke.github.GHRepository;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+
+import static io.xstefank.wildfly.bot.model.RuntimeConstants.DEFAULT_COMMIT_MESSAGE;
+import static io.xstefank.wildfly.bot.utils.Constants.INVALID_COMMIT_MESSAGE;
+import static io.xstefank.wildfly.bot.utils.Constants.PR_JSON;
+import static io.xstefank.wildfly.bot.utils.Constants.VALID_COMMIT_MESSAGE;
+import static io.xstefank.wildfly.bot.utils.Constants.XSTEFANK_REPO;
+
+/**
+ * Tests for the Wildfly -> Format -> Commit checks.
+ */
+@QuarkusTest
+@GitHubAppTest
+public class PRCommitCheckTest {
+
+    private static String wildflyConfigFile;
+    private static GitHubJson gitHubJson;
+    private static String commitMessage;
+
+    @Test
+    void incorrectCommitCheckFailTest() throws IOException {
+        wildflyConfigFile = """
+            wildfly:
+              format:
+                commit:
+                  enabled: true
+            """;
+        gitHubJson = GitHubJson.builder(PR_JSON).build();
+        commitMessage = INVALID_COMMIT_MESSAGE;
+
+        GitHubAppTesting.given()
+                .github(mocks -> Util.prepareMockedRepo(mocks, wildflyConfigFile, gitHubJson, commitMessage))
```

Comment:
@The-Huginn is preparing a better builder instead of this Util method. The whole `prepareMockedRepo` might be removed in favor of the builder, which should have multiple commit messages.